crates/plasma/src/source.rs

```diff
@@ -89,14 +89,14 @@ where
                 Ok(d) => d,
                 Err(e) => {
                     tracing::warn!("failed to pull next data from the plasma source iterator");
-                    return Some(Err(StageError::Plasma(e)));
+                    return Some(Err(StageError::Custom(anyhow!(e))));
                 }
             };
 
             // If the data is empty,
             if data.is_empty() {
                 tracing::warn!("empty data from plasma source");
-                return Some(Err(StageError::Plasma(PlasmaError::NotEnoughData)));
+                return Some(Err(StageError::Custom(anyhow!(PlasmaError::NotEnoughData))));
```

Comment:
Ugh I really dislike `StageError` leaking it's way out of the kona-derive crate. We should fix this in a follow up issue.

Reply:
Gotcha! Yeah, totally open to make it not leak out if it's possible. 

This is also basically a substitute component of the derivation pipeline, so maybe it makes sense?